### ISSUE TITLE
[amplify_secure_storage] SecurityFrameworkError code: -25308

### ISSUE THREAD

**Issue by asaarnak** [aws-amplify/amplify-flutter#2535]
I couldn't reproduce it myself.
Noticed this issue for 2 users on iOS crash logs.
Maybe related to this: https://developer.apple.com/forums/thread/114542
Seems that app is locked maybe and app is on background that`s why this error happens.
https://stackoverflow.com/questions/48068325/keychain-access-error-25308-errsecinteractionnotallowed 

```stacktrace
SecureStorageException(
  
  message: An unknown exception occurred.,
  
  recoverySuggestion: An unknown exception occurred. Please take a look at https://github.com/aws-amplify/amplify-flutter/issues to see if there are any existing issues that match your scenario, and file an issue with the details of the bug if there isn't.,
  
  underlyingException: SecurityFrameworkError: {code: -25308, message: No error string is available.}),
  
  stackTrace=
  
0  ???                            0x0 AmplifySecureStorageCupertino._add + 146 (amplify_secure_storage_cupertino.dart:146)
1  ???                            0x0 AmplifySecureStorageCupertino.write.<fn> + 65 (amplify_secure_storage_cupertino.dart:65)
2  ???                            0x0 (null).using + 124 (arena.dart:124)
3  ???                            0x0 AmplifySecureStorageCupertino.write + 63 (amplify_secure_storage_cupertino.dart:63)
4  ???                            0x0 AmplifySecureStorageDartMixin.write + 42 (amplify_secure_storage_mixin.vm.dart:42)
5  ???                            0x0 SecureStorageWorker.run + 73 (secure_storage_worker.dart:73)
6  ???                            0x0 (null)._run + 14 (secure_storage_worker.worker.vm.dart:14)
```

Seems to be related stacktrace:
```
0  ???                            0x0 CredentialStoreStateMachine.getCredentialsResult + 59 (credential_store_state_machine.dart:59)
1  ???                            0x0 FetchAuthSessionStateMachine.onFetchAuthSession + 179 (fetch_auth_session_state_machine.dart:179)
2  ???                            0x0 FetchAuthSessionStateMachineBase.resolve + 37 (fetch_auth_session_state_machine_base.dart:37)
3  ???                            0x0 StateMachine._listenForEvents + 171 (state_machine.dart:171)
```

**Comment by Jordan-Nelson** [comment]
@asaarnak - I believe the cause of this is that keychain items are being stored with the access `kSecAttrAccessibleWhenUnlockedThisDeviceOnly`, which prevents items from being accessed while the device is locked. This is a bug. Items should be stored with the access `kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly`, which prevents access after a restart until the device has been unlocked.

I have started working on a fix for this.

**Comment by Jordan-Nelson** [comment]
A fix for this has been merged and will be included in the next release.

Since the default value for `kSecAttrAccessible` has changed, this will require end users to re-authenticate if they are upgrading from a version of the app that was using an earlier version of dev-preview.

**Comment by asaarnak** [comment]
Thanks!
Could this be released as hot-fix release?

**Comment by dnys1** [comment]
Yes, absolutely. We'll do a release on Monday.

In the meantime, if you'd like to try the fix, you can add a dependency override in your `pubspec.yaml`:

```yaml
dependency_overrides:
  amplify_secure_storage_dart:
    git:
      url: https://github.com/aws-amplify/amplify-flutter
      ref: next
      path: packages/secure_storage/amplify_secure_storage_dart
```

**Comment by dnys1** [comment]
This fix has been released in v0.2.0 of the secure storage plugins. You should be able to run `flutter pub upgrade` to get the versions now. Thanks for your patience resolving the issue and let us know if you experience any more problems.

**Comment by asaarnak** [comment]
Thanks!
It was quite rare bug, thanks for fixing it!
I tested this and as you said it did do logout and login worked after that.
